paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/sources/PaimonMicroBatchStream.scala

```diff
@@ -43,12 +43,7 @@ class PaimonMicroBatchStream(
     val initSnapshotId = Math.max(
       table.snapshotManager().earliestSnapshotId(),
       streamScanStartingContext.getSnapshotId)
-    val scanSnapshot = if (initSnapshotId == streamScanStartingContext.getSnapshotId) {
```

Comment:
e.g. assume snapshots in FS is: 2, 3, 4 (earliestSnapshotId = 2) 
snapshotId in streamScanStartingContext = 1 and ScanFullSnapshot is true
Should we scan full snapshot for '2' at this time?